test/typecheck/union.c

```diff
@@ -26,5 +26,24 @@ int main() {
 
   s.circle;
 
+  union gender {
+    int male;
+    int female;
+  } unknown_sex;
+
+  union gender {
+    int male;
+    int female;
+  } girl = {
+    .female = 1,
+  };
```

Comment:
I guess the redefinition here is the _TODO_ you mentioned above, am I correct?

Reply:
I was thinking that the TODO would fix this example, but after checking Clang https://gcc.godbolt.org/z/scj7fzsd8. It shouldn't be a correct definition. 😆 I'll remove that TODO comment.
```
struct gender {
   int male;
   int female;
};

union gender {
  int male;
  int female;
};
```

Reply:
I see, then this is a separate issue where the test case itself redefines `gender` (see: <https://godbolt.org/z/v8Paxcqhn>). Even when a custom type is defined alongside a declaration, it introduces a new identifier, which cannot be redefined.

Reply:
> I see, then this is a separate issue where the test case itself redefines `gender` (see: https://godbolt.org/z/v8Paxcqhn). Even when a custom type is defined alongside a declaration, it introduces a new identifier, which cannot be redefined.

You're right! I've removed the redefined part in the test.